retrofit/src/main/java/retrofit2/ParameterHandler.java

```diff
@@ -55,8 +56,20 @@
   }
 
   static final class RelativeUrl extends ParameterHandler<Object> {
+    private final Method method;
+    private final int p;
+
+    RelativeUrl(Method method, int p) {
+      this.method = method;
+      this.p = p;
+    }
+
     @Override void apply(RequestBuilder builder, @Nullable Object value) {
-      checkNotNull(value, "@Url parameter is null.");
+      try {
+        checkNotNull(value, "@Url parameter is null.");
+      } catch (NullPointerException npe) {
```

Comment:
There's no need to do this. We can just check for null locally and pass null as the cause to the `Utils` method.

Reply:
Yeaaah, I was going to change this. Realized this when commenting before and reviewing the code. Just didn't access to my computer then. I'll work on this and the other changes right now.